src/modules/ask_to_visit.rs

```diff
@@ -0,0 +1,68 @@
+use std::sync::Arc;
+
+use anyhow::Result;
+use diesel::{ExpressionMethods, QueryDsl, RunQueryDsl};
+use teloxide::requests::Requester;
+use teloxide::types::{ChatId, Message};
+use teloxide::Bot;
+
+use crate::common::{BotEnv, UpdateHandler};
+use crate::db::DbUserId;
+use crate::schema;
+
+pub fn message_handler() -> UpdateHandler {
+    dptree::entry().branch(
+        dptree::filter(|env: Arc<BotEnv>, msg: Message| {
+            env.config.telegram.chats.ask_to_visit.has_message(&msg)
+        })
+        .endpoint(handle_message),
+    )
+}
+
+async fn handle_message(
+    bot: Bot,
+    env: Arc<BotEnv>,
+    msg: Message,
+) -> Result<()> {
+    log::info!("Got message {:?}", msg);
+    let Some(text) = msg.text() else {
+        return Ok(());
+    };
+    if text.starts_with("//") {
+        return Ok(());
+    };
+    let Some(from) = msg.from else { return Ok(()) };
+
+    if let Some(data) = &*env.active_macs.read().await {
```

Comment:
None is set when the bot starts, and then the mac_monitoring function is run to populate the values. Option is needed to detect the state when the bot has just started and there is no data in it, so that immediately after startup it does not send a message that all already present macs are online.

Reply:
Fixed in 4ecd3d12ece6f77d149c561b490d2e75f9582b46